### PR TITLE
Improve `~course` command

### PR DESCRIPTION
**Pull request by teppah**
This PR: 
- adds a Prettier config file for more consistent formatting
- cleans up the way the GraphQL request is made by replacing `node-fetch` with `graphql-request`
- adds clickable course links to the prerequisites and antirequisites sections in the Discord embed

### CODE REVIEW

File: commands/course.js [hunk 1]
```diff
@@ -1,41 +1,69 @@
-const Discord = require('discord.js');
-const fetch = require('node-fetch');
+const Discord = require("discord.js");
+const { request, gql } = require("graphql-request");
+const { uwflowQuery, uwflowEndpoint } = require("../util/uwflow-constants");
+
+const courseMatcher = /([a-zA-Z]{2,}[ ]?\d+[a-zA-Z]?)/g;
```
Comment (sunny-zuo) [inline commands/course.js:5]: I believe that the courseMatcher regex is declared but never used; is this intentional?

Comment (teppah) [reply commands/course.js:5]: See lines 40-43:
```javascript
const prereqs =
    course.prereqs?.replaceAll(courseMatcher, replacer) ?? "None";
const antireqs =
    course.antireqs?.replaceAll(courseMatcher, replacer) ?? "None";
```

Comment (sunny-zuo) [reply commands/course.js:5]: My bad, I think I was looking at the wrong commit. Thanks.

**Comment by sunny-zuo** [comment]
I've replaced all instances of `replaceAll` with `replace`, using global regex as necessary to maintain functionality.  `replaceAll` is only available starting in [Node 15.x](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll), and I would prefer to keep running Node LTS (14.x).

Besides that, everything looks good! There does seem to be one edge case I've noticed with linking. Running `$course ece106` outputs this for prerequisites:

> [ECE105](https://uwflow.com/course/ece105); Level at [least 1B](https://uwflow.com/course/least1b) Computer Engineering or Electrical Engineering or Software Engineering.

There's probably a decent number of edge cases that would be difficult to handle in an easy way, so I'm fine with merging this PR in its current state. Let me know if you'd like to attempt a fix, or if I should merge.

**Comment by teppah** [comment]
Neat, I didn't know passing a regex with the `g` flag to `replace()` overrides the default behavior of only replacing the first occurrence when a pattern string is passed.

As for the edge cases, I've just added a regex to test for disallowed matches such as the word `least`. Maybe that regex could be updated in the future as we find more of them.

**Comment by sunny-zuo** [comment]
Sounds good!
